awx/main/fields.py

```diff
@@ -740,8 +739,24 @@ def validate(self, value, model_instance):
 
         class TowerNamespace:
             filename = None
-
         valid_namespace['tower'] = TowerNamespace()
+
+        # ensure either single file or multi-file syntax is used (but not both)
+        template_names = set(key for type_, injector in value.items()
+                             for key, tmpl in injector.items()
+                             if key.startswith('template'))
+        if 'template' in template_names and len(template_names) > 1:
+            raise django_exceptions.ValidationError(
+                _('Must use multi-file syntax when injecting multiple files'),
+                code='invalid',
+                params={'value': value},
+            )
+        if 'template' not in template_names:
+            valid_namespace['tower'].filename = TowerNamespace()
+            for template_name in template_names:
+                template_name = template_name.split('.')[1]
+                setattr(valid_namespace['tower'].filename, template_name, 'EXAMPLE')
```

Comment:
Does this validate if the user uses syntax like `"{{tower.filename.cert}"`, where the cert file isn't specified, and is that even possible to check?

Reply:
@AlanCoding - Good question - the short answer is that lines 760-771 (in the new version of this file) will throw a `ValidationError` if a file template is not defined. 

The slightly longer answer is.. in the section I mentioned, lines 763-765 simulate rendering a template using information in a namespace. If `tower.filename.cert` is not in the namespace, then `ValidationError` will be raised when `render` is called. Lines 754-758 create the namespace (using `EXAMPLE` in place of the actual file template). So, in short, we simulate rendering extravars, environment variables, and file templates in order to confirm that all the variables they require exist.

Reply:
👍 this makes sense to me, and uses the same validation approach as the old single `template` usage.